myia/operations/macro_to_scalar_type.py

```diff
@@ -0,0 +1,63 @@
+"""
+Implementation of 'to_scalar_type' operation.
+
+Convert given data into an appropriate abstract scalar.
+If data is already an abstract type, return the result of method xvalue().
+If data is already an abstract scalar with a number type, return it as is.
+If data is an abstract scalar with a string type, try to infer type
+from given string and return an abstract scalar with inferred type.
+Used in myia/operations/op_full.py
+"""
+
+import numpy as np
+
+from ..lib import (
+    ANYTHING,
+    TYPE,
+    VALUE,
+    AbstractScalar,
+    AbstractType,
+    Constant,
+    macro,
+)
+from ..xtype import Nil, Number, String, pytype_to_myiatype
+
+
+def convert_to_scalar_type(sync_data, default=None):
+    """
+    Main code to convert given data to abstract scalar.
+
+    :param sync_data: arbitrary data to convert
+    :param default: optional default value to return.
+        If not provided and if sync_data cannot be converted,
+        then an exception is raised.
+    :return: an abstract scalar object if data can be converted,
+        or default value if provided, otherwise raise an exception.
+    """
+    if isinstance(sync_data, AbstractType):
+        return Constant(sync_data.xvalue())
+    if isinstance(sync_data, AbstractScalar):
```

Comment:
This is mostly a style convention, but I find it preferable to use `elif` here. It's a bit easier to read (you can immediately see they are mutually exclusive branches), and it behaves better if for some reason you forget a `return` statement somewhere.

Reply:
Done!